compiler/src/dotty/tools/dotc/core/Annotations.scala

```diff
@@ -87,6 +87,19 @@ object Annotations {
     def sameAnnotation(that: Annotation)(using Context): Boolean =
       symbol == that.symbol && tree.sameTree(that.tree)
 
+    def hasOneOfMetaAnnotation(metaSyms: Symbol*)(using Context): Boolean =
+      def recTp(tp: Type): Boolean = tp.dealiasKeepAnnots match
+        case AnnotatedType(parent, metaAnnot) => metaSyms.exists(metaAnnot.matches) || recTp(parent)
+        case _ => false
+      def rec(tree: Tree): Boolean = methPart(tree) match
+        case New(tpt) => rec(tpt)
+        case Select(qual, _) => rec(qual)
+        case Annotated(arg, metaAnnot) => metaSyms.exists(metaAnnot.tpe.classSymbol.derivesFrom) || rec(arg)
```

Comment:
Why does this one use `derivesFrom` but the other cases don't?

Reply:
The other ones are on the available Annotation, which here isn't available.  And Annotation#matches is:

```scala
def matches(cls: Symbol)(using Context): Boolean = symbol.derivesFrom(cls)
```

Reply:
OK, makes sense.